pkg/ddc/thin/transform_config.go

```diff
@@ -18,14 +18,30 @@ package thin
 
 import (
 	"encoding/json"
+	"fmt"
+	"strings"
 
 	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/utils/kubeclient"
+	corev1 "k8s.io/api/core/v1"
 )
 
 func (t *ThinEngine) transformConfig(runtime *datav1alpha1.ThinRuntime,
 	dataset *datav1alpha1.Dataset, targetPath string) (config Config, err error) {
 	mounts := []datav1alpha1.Mount{}
+	pvAttributes := map[string]*corev1.CSIPersistentVolumeSource{}
```

Comment:
Please add to do for flex volume. I suggest finding a more flexible way rather than using corev1.CSIPersistentVolumeSource.